src/sso/pages/Login.tsx

```diff
@@ -35,8 +34,18 @@ function Login() {
   const [showQR, setShowQR] = useState<string>();
   const navigation = useNavigate();
   const communication = useCommunicationStore((state) => state.communication);
+  let rendered = false;
 
   useEffect(() => {
+    // Prevent useEffect from running twice which causes a race condition of the
+    // async handleRequests() between which publicKey is sent in the request and this
+    // conflicts in the publicKey that is saved in localStorage
+    if (!rendered) {
+      rendered = true;
+    } else {
+      return;
+    }
+
```

Comment:
on production side this wont run twice it was made to make sure if the user refreshes twice so fast the correct behavior will still be called.
this wont fix that issue but rather will just make sure the development looks like production

Reply:
on production this would not be an issue but we need a fix for the development environment so devs can test. It's not about the user refreshing, but React rendering twice. Do you have a suggestion of what to do here?